src/commands/mod.rs

```diff
@@ -1024,22 +1026,42 @@ fn cmd_config(
     ui.request_pager();
     let settings = command.settings();
     match subcommand {
-        ConfigSubcommand::List { name } => {
-            let raw_values = match name {
-                Some(name) => {
-                    settings
-                        .config()
-                        .get::<config::Value>(name)
-                        .map_err(|e| match e {
-                            config::ConfigError::NotFound { .. } => {
-                                user_error("key not found in config")
-                            }
-                            _ => e.into(),
-                        })?
+        ConfigSubcommand::List {
+            name,
+            include_defaults,
+        } => {
+            let name_path = name
+                .as_ref()
+                .map_or(vec![], |name| name.split('.').collect_vec());
+            let values = command.resolved_config_values(&name_path)?;
```

Comment:
Maybe just make `values` mutable and add:

```rust
values.sort_unstable_by(|a, b| a.path.cmp(&b.path));
```

so that values are printed in order.

Reply:
Oh, but at this level sorting the entire list would be a misfeature. It's sorted by (layer, key) and I'd like to not disturb that existing ordering by logical layer.

Reply:
I understand that, but why is it useful? For example, here is my `jj config list` for the current repository:

```
ui.editor="vim"
ui.graph.style="curved"
user.email="sam@rfc1149.net"
user.name="Samuel Tardieu"
git.fetch="upstream"
```

The `git.fetch` feels oddly (to me) out of order at the end. It is there because it comes from the repository configuration instead of the user one, but nothing indicates that.

But this can always be discussed and fixed if needed later if you want to get those changes in.

Reply:
FWIW, Mercurial sorts them just by key name, and Git sorts them by source and then by key name. I don't know which I would prefer.

Reply:
Right, every tool seems to do it differently, not sure there's any clearly "superior" ordering as long as it's deterministic.

Git orders entirely in order of definition/precedence, which makes the output more intuitive for options like its `--show-origin` and when it's listing overridden values. Mercurial seems to alphabetize it completely at the *top* level, but then keep nested keys in definition order.

Granted, if your config is all in one file except for a single entry having it 4/5 alphabetized looks a little mysterious. Fully alphabetizing would keep some related options close together but also splits up related options like "colors" and "ui". It would also be possible to maintain a logical ordering for known config keys that could sort of correspond to the order they're introduced in https://github.com/martinvonz/jj/blob/main/docs/config.md, but that's complicated and I felt like alphabetizing wasn't actually an improvement over retaining the precedence ordering.

Reply:
K, seems like I can call the ordering question resolved and revisit later if needed, but LMK any other concerns. Will merge a bit later today.